src/Offer/ReadModel/JSONLD/OfferLDProjector.php

```diff
@@ -123,6 +126,7 @@ public function __construct(
         $this->mediaObjectSerializer = $mediaObjectSerializer;
         $this->basePriceTranslations = $basePriceTranslations;
         $this->videoNormalizer = $videoNormalizer;
+        $this->weights = $weights;
```

Comment:
Given that this should be an array of integers, should we either check the inserted weights, or cast them all to integers?

`$this->weights = array_map('intval', $weights);`

Reply:
I will create a class `Weights` that throws on wrong input types. I will do this in a separate pull request.